Ported build system to cmake

Algebro7 opened the pull request:
Sorry to hit you with another PR so soon, but I finished porting the project to optionally use cmake as the build system. Heres how to do it:

1) Clone the repo
2) `cd Pokopom && mkdir build`
3) `cmake ../`

The cmake command will automatically generate a Visual Studio Solution file or GCC Makefile for you in the build directory, depending on your platform. You can open the solution with Visual Studio and build the library. I am doing all of my Windows 7 testing in a VM and I cannot test how well the plugin works or if I made any mistakes, but I have confirmed that PCSX2 sees the plugin and can configure it normally.

Commit:
Ported build system to cmake

Commit:
Fixed project name

KrossX commented on CMakeLists.txt:
```diff
@@ -0,0 +1,57 @@
+project(Pokopomclion)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_BUILD_TYPE Debug)
```
Would setting this to Release make it the default build type on Visual Studio? That would be preferable.

Algebro7 replied:
I think so. I'll test it and update the commit if it works.

Algebro7 replied:
As per this article https://stackoverflow.com/questions/19024259/how-to-change-the-build-type-to-release-mode-in-cmake, it looks like for Visual Studio you have to specify it when you actually run cmake, like this:
```cmake --build ../ --target ALL_BUILD --config Release```
I can provide these instructions somewhere if you'd like, or include a batch file. Also, building with the original VS solution file rather than cmake still works if users prefer that.

I really appreciate your help reviewing these changes!

KrossX commented on CMakeLists.txt:
```diff
@@ -0,0 +1,57 @@
+ADD_DEFINITIONS(-DUNICODE)
+ADD_DEFINITIONS(-D_UNICODE)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -m32 -D_UNICODE -DUNICODE")
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
```
Does this really mean it would export all sysmbols? Only those in Exports.def should be exported.

Algebro7 replied:
Good point, let me look at how to do this with cmake

Algebro7 committed:
Changed build-type to release; Export only Exports.def now

Build-type is now release, but this only affects makefiles. Visual Studio projects will still have to be modified to release from inside VS. Also specified the symbols file as Exports.def, so only those symbols are exported now.

Algebro7 commented:
Thanks again for the review, I've fixed the issues you mentioned in the most recent commit.

KrossX commented:
Alright!

halian commented:
When building Pokopom this way under Arch Linux, I get an unusable plugin, with PCSX2 throwing `[wx] /usr/lib/pcsx2/libpadPokopom.so: wrong ELF class: ELFCLASS32` on load.

KrossX commented:
This build makes a 32bit plugin, so unless you use it with a 32bit program, it will fail.
